src/fiatExchanges/quotes/FiatConnectQuote.ts

```diff
@@ -152,12 +160,70 @@ export default class FiatConnectQuote extends NormalizedQuote {
     return this.quoteResponseKycSchema?.kycSchema
   }
 
-  // TODO: Dynamically generate time estimation strings
-  getTimeEstimation(): SettlementTime {
-    // payment method can only be bank or fc mobile money
-    return this.getPaymentMethod() === PaymentMethod.Bank
-      ? SettlementTime.ONE_TO_THREE_DAYS
-      : SettlementTime.LESS_THAN_24_HOURS
+  getTimeEstimation(): SettlementEstimation {
+    const fiatAccountInfo = this.quote.fiatAccount[this.getFiatAccountType()]
+    if (!fiatAccountInfo?.settlementTimeUpperBound) {
+      // payment method can only be bank or fc mobile money
+      return this.getPaymentMethod() === PaymentMethod.Bank
+        ? DEFAULT_BANK_SETTLEMENT_ESTIMATION
+        : DEFAULT_MOBILE_MONEY_SETTLEMENT_ESTIMATION
+    }
+
+    const daysToSettlementEstimation = (
+      lowerBound: number,
+      upperBound: number
+    ): SettlementEstimation => {
+      if (lowerBound === upperBound || lowerBound === 0) {
+        return {
+          settlementTime: SettlementTime.LESS_THAN_X_DAYS,
+          upperBound,
+        }
+      }
+
+      return {
+        settlementTime: SettlementTime.X_TO_Y_DAYS,
+        lowerBound,
+        upperBound,
+      }
```

Comment:
it may not be worth the effort, but this does look almost identical between the two functions. There's probably a way to make it more DRY

Reply:
I'm not that concerned about it though

Reply:
Was struggling to cleanly make it more DRY, so am sticking with this for now